### PR TITLE
Add GaussianDropout

### PR DESCRIPTION
**Pull request by the-moliver**
Now that the noise module is official I wanted to push my original noise layer addition of Gaussian Dropout, which is multiplicative (rather than additive) Gaussian Noise.

### CODE REVIEW

File: keras/layers/noise.py [hunk 1]
```diff
@@ -20,4 +20,26 @@ def get_output(self, train=False):
...
+    '''
+    def __init__(self, p):
+        super(GaussianDropout,self).__init__()
+        self.p = p
+
+    def get_output(self, train):
+        X = self.get_input(train)
+        if train:
+            # self.p refers to drop probability rather than retain probability (as in paper) to match Dropout layer syntax
+            X *= srng.normal(size=X.shape, avg = 1., std=T.sqrt(self.p/(1-self.p)), dtype=theano.config.floatX)
```
Comment (fchollet) [inline keras/layers/noise.py]: > `T.sqrt(self.p/(1-self.p))`

Something should be done to avoid division by zero in case p = 1... maybe clipping p to [epsilon, 1-epsilon]?

In terms of coding style, please use spaces around operators and use floats in float operations.

**Comment by the-moliver** [comment]
Added the spaces and changed 1 to 1.0. This is meant to be a drop-in replacement for a dropout layer as in the paper, i.e. `Dropout(.5)` could be changed to `GaussianDropout(.5)`. So in either case `p=1.0` doesn't really make sense and would break things.

**Comment by simonhughes22** [comment]
Can you update the module with a link to the appropriate paper(s)?

**Comment by the-moliver** [comment]
That is the appropriate paper! Read it! ;-)

**Comment by simonhughes22** [comment]
Sorry I was referring to the regular GaussianLayer. Also, a url would be good.

**Comment by the-moliver** [comment]
URL added for GaussianDropout

**Comment by simonhughes22** [comment]
Awesome, thanks.
